### PR TITLE
Fix: replace v-bind with object prop passing

### LINKED ISSUE #12489: [vue3]: do not use v-bind on FilePreview
follow-up for Nc30: not sure yet whether and how broken it might be with Vue3, but would be more stable to migrate it to a single object prop

- Ref: https://github.com/nextcloud/spreed/pull/12395

_Originally posted by @Antreesy in https://github.com/nextcloud/spreed/issues/12467#issuecomment-2157720783_

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #12489

### 🏁 Checklist

- [ ] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.spec.js [hunk 1]
```diff
@@ -51,14 +51,16 @@ describe('FilePreview.vue', () => {
...
-			previewAvailable: 'yes',
+			file: {
+				id: '123',
+				name: 'test.jpg',
+				path: 'path/to/test.jpg',
+				size: '128',
+				etag: '1872ade88f3013edeb33decd74a4f947',
+				permissions: '15',
+				mimetype: 'image/jpeg',
+				previewAvailable: 'yes',
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.spec.js:62]:
```suggestion
				'preview-available': 'yes',
```
and one more in `test('renders generic mime type icon for unknown mime types'`

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.spec.js [hunk 2]
```diff
@@ -96,7 +98,7 @@ describe('FilePreview.vue', () => {
 		})
 
 		test('renders file preview for guests', async () => {
-			propsData.link = 'https://localhost/nc-webroot/s/xtokenx'
+			propsData.file.link = 'https://localhost/nc-webroot/s/xtokenx'
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.spec.js:101]: check all `propsData.` changes in tests, some are missing adding a `file` like here

**Review by ShGKme — APPROVED** [review]
Thanks

**Review by Antreesy — APPROVED** [review]
Tested

**Pull request merged**